src/client_side_request.cc

```diff
@@ -1608,10 +1606,17 @@ ClientHttpRequest::sslBumpStart()
         return;
     }
 
+    auto establishedReply = new HttpReply;
+    establishedReply->sline.set(Http::ProtocolVersion(), Http::scOkay, "Connection established");
+    assert(!al->reply);
+    al->reply = establishedReply;
+    HTTPMSGLOCK(al->reply)
+
+    const auto mb = establishedReply->pack();
     // send an HTTP 200 response to kick client SSL negotiation
     // TODO: Unify with tunnel.cc and add a Server(?) header
-    static const char *const conn_established = "HTTP/1.1 200 Connection established\r\n\r\n";
-    Comm::Write(getConn()->clientConnection, conn_established, strlen(conn_established), bumpCall, NULL);
+    Comm::Write(getConn()->clientConnection, mb->content(), mb->contentSize(), bumpCall, nullptr);
```

Comment:
If you have not already, please check that this Comm::Write() will immediately copy the bytes passed to it.

Reply:
No, it does not copy. I took this example from errorSend() (errorpage.cc). It looks like that code has the same mistake.

Reply:
> It looks like that code has the same mistake.

No, AFAICT, that code uses a different form of Comm::Write() that is free from this bug due to `MemBuf::stolen hack`. If possible, do use that (simpler-looking) Comm::Write() call here as well.